src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/StaffRefDataServiceImpl.java

```diff
@@ -1,53 +1,288 @@
+import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.PROFILE_ALREADY_CREATED;
+import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.ROLE_CWD_USER;
+import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.ROLE_STAFF_ADMIN;
+import static uk.gov.hmcts.reform.cwrdapi.util.CaseWorkerConstants.SRD;
 
 @Service
 @Slf4j
 @SuppressWarnings("AbbreviationAsWordInName")
 public class StaffRefDataServiceImpl implements StaffRefDataService {
 
+
     @Value("${loggingComponentName}")
     private String loggingComponentName;
 
+    @Value("${crd.publisher.caseWorkerDataPerMessage}")
+    private int caseWorkerDataPerMessage;
+
+    @Autowired
+    CaseWorkerProfileRepository caseWorkerProfileRepo;
+
+    @Autowired
+    RoleTypeRepository roleTypeRepository;
+
+    @Autowired
+    UserTypeRepository userTypeRepository;
+
+    @Autowired
+    private UserProfileFeignClient userProfileFeignClient;
+
     @Autowired
-    private UserTypeRepository userTypeRepository;
+    private TopicPublisher topicPublisher;
+
+    @Autowired
+    IValidationService validationServiceFacade;
 
     @Autowired
     SkillRepository skillRepository;
 
+    @Autowired
+    IJsrValidatorInitializer validateStaffProfile;
 
     @Autowired
-    RoleTypeRepository roleTypeRepository;
+    StaffProfileCreateUpdateUtil staffProfileCreateUpdateUtil;
+
+
+    @SuppressWarnings("unchecked")
+    public StaffProfileCreationResponse processStaffProfileCreation(StaffProfileCreationRequest staffProfileRequest) {
+
+        final CaseWorkerProfile newStaffProfiles;
+        final CaseWorkerProfile processedStaffProfiles;
+        final StaffProfileCreationResponse response;
+
+        validateStaffProfile.validateStaffProfile(staffProfileRequest);
+
+        checkStaffProfileEmailAndSuspendFlag(staffProfileRequest);
+        newStaffProfiles = createCaseWorkerProfile(staffProfileRequest);
+
+        processedStaffProfiles = persistStaffProfile(newStaffProfiles,staffProfileRequest);
+
+        response = StaffProfileCreationResponse.builder()
+                    .caseWorkerId(processedStaffProfiles.getCaseWorkerId())
+                    .build();
+
+        return response;
+    }
+
+
+
+    private void checkStaffProfileEmailAndSuspendFlag(StaffProfileCreationRequest profileRequest) {
+
+        // get all existing profile from db (used IN clause)
+        CaseWorkerProfile dbCaseWorker = caseWorkerProfileRepo.findByEmailId(profileRequest.getEmailId());
+
+        if (isNotEmpty(dbCaseWorker)) {
+            invalidRequestError(profileRequest, PROFILE_ALREADY_CREATED);
+        }
+
+        if (profileRequest.isSuspended()) {
+            invalidRequestError(profileRequest, NO_USER_TO_SUSPEND_PROFILE);
+        }
+    }
+
+    private void invalidRequestError(StaffProfileCreationRequest profileRequest, String errorMessage) {
+
+        validationServiceFacade.saveStaffAudit(AuditStatus.FAILURE,NO_USER_TO_SUSPEND_PROFILE,
```

Comment:
I think you need to pass "errorMessage" instead of "NO_USER_TO_SUSPEND_PROFILE,"